flatfs.go

```diff
@@ -691,33 +728,37 @@ func folderSize(path string, deadline time.Time) (int64, error) {
 	du += duEstimation
 	du += stat.Size()
 	//fmt.Println(path, "total:", totalFiles, "totalStat:", i, "totalFile:", filesProcessed, "left:", nonProcessed, "avg:", int(avg), "est:", int(duEstimation), "du:", du)
-	return du, nil
+	return du, accuracy, nil
 }
 
 // calculateDiskUsage tries to read the DiskUsageFile for a cached
 // diskUsage value, otherwise walks the datastore files.
+// it is only safe to call in Open()
 func (fs *Datastore) calculateDiskUsage() error {
 	// Try to obtain a previously stored value from disk
 	if persDu := fs.readDiskUsageFile(); persDu > 0 {
-		atomic.StoreInt64(&fs.diskUsage, persDu)
+		fs.diskUsage = persDu
```

Comment:
Eventually, I'd like to *also* mark the on-file disk usage as an estimate until we close. That way, if we crash, we know that our estimate may be off by a bit. However, we can probably do that in a future PR (your preference).